cargo-process.el

```diff
@@ -487,7 +487,9 @@ Meant to be run as a `compilation-filter-hook'."
              (rust-beginning-of-defun))
             ((fboundp 'rustic-beginning-of-defun)
              (rustic-beginning-of-defun))
-            (t (user-error "%s needs either rust-mode or rustic-mode"
+            ((fboundp 'treesit-beginning-of-defun)
+             (treesit-beginning-of-defun))
+            (t (user-error "%s needs either rust-mode, rustic-mode, or rust-ts-mode"
                            this-command))))
     (beginning-of-line)
     (search-forward "fn ")
```

Comment:
instead of this, would it be more robust (and more elegant) to find the function name with something like this:
```elisp
 (re-search-forward "fn\\s +\\(.*\\_>\\)") 
 (match-string 1)
```
Additionally, the defun does not return the current test, it returns the name of the current function.

Reply:
I mean instead of the `(search-forward "fn ")` part